cartography/intel/aws/iam.py

```diff
@@ -336,6 +337,64 @@ def load_user_access_keys(session, user_access_keys, aws_update_tag):
                 )
 
 
+def evaluate_trust_policies(neo4j_session):
+    query = """
+    MATCH (origin:AWSAccount)-[:RESOURCE|AWS_ROLE]->()-[r:STS_ASSUMEROLE_ALLOW]->(role:AWSRole)
+        <-[:AWS_ROLE]-(account:AWSAccount)
+    WHERE account <> origin AND NOT exists(
+        (role)-[:TRUSTS_AWS_PRINCIPAL]->(:AWSPrincipal{arn: 'arn:aws:iam::' + origin.id + ':root'})
+    )
+    WITH r
+    DELETE r;
+    """
+    neo4j_session.run(query)
+
+
+def expand_role_wildcards(neo4j_session, update_tag):
+    wildcard_role_query = """
+    MATCH (role:AWSRole)
+    WHERE role.arn CONTAINS '*'
+    RETURN role.arn;
+    """
+
+    absolute_role_query = """
+    MATCH (role:AWSRole)
+    WHERE NOT role.arn CONTAINS '*'
+    RETURN role.arn;
+    """
+
+    query = """
+    MATCH (origin:AWSAccount)-[:AWS_ROLE]->(assumer:AWSRole)
+        -[:STS_ASSUMEROLE_ALLOW]->(wildcard:AWSRole{arn: {WildcardArn}})
+    WITH origin, assumer, wildcard
+    MATCH (absolute:AWSRole{arn: {AbsoluteArn}})
+    WHERE exists( (absolute)-[:TRUSTS_AWS_PRINCIPAL]->(:AWSPrincipal{arn: 'arn:aws:iam::' + origin.id + ':root"}) )
```

Comment:
Using the reasoning in my earlier comment about partial trusts, I think we can remove this `where exists` clause so that we wildcard expand to all computed absolute roles (not just the ones that have trusts back to the origin account's root principal).